src/Nethermind/Nethermind.Optimism.Test/OptimismWithdrawalTests.cs

```diff
@@ -0,0 +1,134 @@
+// SPDX-FileCopyrightText: 2025 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System.Collections.Generic;
+using FluentAssertions;
+using Nethermind.Core;
+using Nethermind.Core.Crypto;
+using Nethermind.Core.Extensions;
+using Nethermind.Core.Specs;
+using Nethermind.Core.Test.Builders;
+using Nethermind.Db;
+using Nethermind.Int256;
+using Nethermind.Logging;
+using Nethermind.State;
+using Nethermind.Trie.Pruning;
+using NSubstitute;
+using NUnit.Framework;
+
+namespace Nethermind.Optimism.Test;
+
+public class OptimismWithdrawalTests
+{
+    /// <summary>
+    /// The value that the actual storage root of <see cref="PreDeploys.L2ToL1MessagePasser"/> will be calculated to.
+    /// </summary>
+    private static readonly Hash256 ActualStorageRoot = new("0xf38f9f63c760d088d7dd04f743619b6291f63beebd8bdf530628f90e9cfa52d7");
+
+    [TestCaseSource(nameof(WithdrawalsRootData))]
+    public void Withdrawals_Processing(ulong timestamp, Hash256? withdrawalHash)
```

Comment:
Test case could have a better name. It's not clear exactly what it's trying to test.